src/parse/response.rs

```diff
@@ -91,15 +91,32 @@ pub fn textstring(input: &[u8]) -> IResult<&[u8], &str> {
 
 /// Reply-line = *( Reply-code "-" [ textstring ] CRLF )
 ///                 Reply-code [ SP textstring ] CRLF
-pub fn Reply_line(input: &[u8]) -> IResult<&[u8], &[u8]> {
-    let parser = tuple((
-        many0(tuple((Reply_code, tag(b"-"), opt(textstring), CRLF))),
-        Reply_code,
-        opt(tuple((SP, textstring))),
-        CRLF,
-    ));
+pub fn Reply_lines(input: &[u8]) -> IResult<&[u8], Response> {
```

Comment:
Urgh... I see your point. Not sure if it is better to keep the (confusing) name or diverge from the RFC. I'll let you decide :-)

Reply:
My proposal would be to ditch the "direct" interface in `parser` and instead provide a `Response::decode()` or `Response::from_bytes()` constructor (but not as part of this PR.